src/daemon/globals.c

```diff
@@ -0,0 +1,13 @@
+#include "globals.h"
+#include "plugin.h"
+/*
+ * Global variables
+ */
+char hostname_g[DATA_MAX_NAME_LEN];
+const int hostname_g_size = sizeof (hostname_g);
```

Comment:
I'm not quite happy about the introduction of this variable. How about we change `hostname_g` to be a `char *` instead? I think it would also be a good idea to add a simple setter function:

```c
/* hostname_set updates hostname_g */
void hostname_set(char const *hostname);
```